src/app/directives/common/disable-focusable-elements/disable-focusable-elements.directive.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Directive, Input, Renderer2, ElementRef, OnChanges,
+} from '@angular/core';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { take, timer } from 'rxjs';
+import { IxSimpleChanges } from 'app/interfaces/simple-changes.interface';
+
+@UntilDestroy()
+@Directive({
+  selector: '[disableFocusableElements]',
+})
+export class DisableFocusableElementsDirective implements OnChanges {
+  @Input() disableFocusableElements: boolean;
+
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+    private renderer: Renderer2,
+  ) {}
+
+  ngOnChanges(changes: IxSimpleChanges<this>): void {
+    if (changes.disableFocusableElements) {
+      this.updateFocusableElements();
+    }
+  }
+
+  private updateFocusableElements(): void {
+    timer(0).pipe(take(1), untilDestroyed(this)).subscribe(() => {
+      const tabIndex = this.disableFocusableElements ? -1 : 0;
+      this.updateTabIndex(tabIndex);
+    });
+  }
+
+  private updateTabIndex(tabIndex: number): void {
+    const focusableElements = this.getFocusableElements();
+    focusableElements.forEach((element) => {
+      this.renderer.setAttribute(element, 'tabindex', tabIndex.toString());
+      this.updateDisabledAttribute(element, tabIndex);
+    });
+  }
+
+  private getFocusableElements(): NodeListOf<HTMLElement> {
+    return this.elementRef.nativeElement.querySelectorAll(
+      'a, button, input, textarea, select, [tabindex]',
+    );
+  }
+
+  private updateDisabledAttribute(element: HTMLElement, tabIndex: number): void {
+    if (tabIndex === -1) {
+      this.renderer.setAttribute(element, 'disabled', 'true');
+    } else if (!element.closest('.role-missing')) {
```

Comment:
Instead of hardcoding class from a different directive, try to apply `DisableFocusableElementsDirective` in `RequiresRolesDirective`, for example by instantiating the class manually and passing in `elementRef` and `renderer`. You may need to call lifecycle hooks manually.